File: GoScootIoS/GoScootIoS/Models/Domain/Bike.swift
//
//  Bike.swift
//  GoScootIoS
//
//  Created by KumikoOumae on 8/11/25.
//

enum BikeType: Codable {
    case scooter, bike
}

struct Bike: Codable, Identifiable {
    let id: String
    let name: String
    let battery_status: Int32
    let maximum_speed: Int32
    let maximum_functional_distance: Int32
    
    enum CodingKeys: String, CodingKey {
        case id, name, battery_status, maximum_speed, maximum_functional_distance
    }
}

File: GoScootAndroid/app/src/main/java/com/example/goscootandroid/Models/Domains/Bike.kt
package com.example.goscootandroid.Models.Domains
import android.annotation.SuppressLint
import kotlinx.serialization.Serializable


@Serializable
enum class BikeType {
    SCOOTER,
    BIKE
}


@SuppressLint("UnsafeOptInUsageError")
@Serializable
data class Bike(
    val id: String,
    val name: String,
    val battery_status: Int,
    val maximum_speed: Int,
    val maximum_functional_distance: Int
)